### PR TITLE
Shows error alert on mobile device 

### LINKED ISSUE #5: Display unsupported message for non-Android mobile users
TODO:

- [x] Try to get the best possible functionality on Android, autoplay will not work, by design.
- [ ] On iOS, and other mobile user agents display a message saying we don't support mobile browsers - since iOS will show videos in full screen, by design. see comments in #38 

Major bugs on Android

screenshot_2015-05-11-07-57-32
screenshot_2015-05-11-07-57-44

### PR DESCRIPTION
**Pull request by FleetAdmiral**
Shows an error alert if user is accessing the website on an mobile device , fixes #5 .

### CODE REVIEW

**Comment by shakeelmohamed** [comment]
Thanks, can we actually do this for all mobile users? I'm not sure a website is even a reasonable means of using ZAP on mobile.

**Comment by FleetAdmiral** [comment]
@shakeelmohamed Are you referring to android and the rest too? I thought the issue persisted only with iOS specific APIs. But i'm more than ok with taking up this task too. So what say , not support any mobile devices at all?

**Comment by shakeelmohamed** [comment]
@FleetAdmiral correct, don't support any devices. Also can we make this a UI change? Around line 429 in `js/everything.js`, inside the onready listener, we can do the user agent check, use jQuery to do something like `$("#container").html("<p>We don't support mobile devices</p>");` then `return;`. The idea is to avoid even attempting to do any of the normal functionality when on a mobile device

**Comment by FleetAdmiral** [comment]
@shakeelmohamed Sorry for being so late , kind of caught up with my own internship . Reading what you've written , i feel it wold be better if we just redirected the user to an error page mentioning/explaining that we don't support mobile , rather than changing the content of the same page. Gives us more space.
What do you say?

File: js/everything.js [hunk 1]
```diff
@@ -474,7 +473,13 @@ $(function() {
             );
         }
     }
-
+    if ( /Android|webOS|iPhone|iPad|iPod|BlackBerry|IEMobile|Opera Mini/i.test(navigator.userAgent) )
```
Comment (shakeelmohamed) [inline js/everything.js]: This needs to be moved up so it's the first thing done in the onready listener, before  the `// Keen.io` line.

File: js/everything.js [hunk 2]
```diff
@@ -474,7 +473,13 @@ $(function() {
...
+    {
+        $("#container").ready(function() {
```
Comment (shakeelmohamed) [inline js/everything.js]: Is this listener necessary? The document should already be ready

File: js/everything.js [hunk 3]
```diff
@@ -474,7 +473,13 @@ $(function() {
...
+          $("#container").html("<p></p>");
```
Comment (shakeelmohamed) [inline js/everything.js]: We shouldn't add a paragraph inside of a header. You can add a `<p>` tag to the HTML that's hidden by default, then here do `$("change-mobile").show();`

File: js/everything.js (hunk 2 above)
Comment (FleetAdmiral) [inline js/everything.js]: @shakeelmohamed Added the listener as a formality , will remove it and make the changes mentioned.

File: js/everything.js [hunk 4]
```diff
@@ -474,7 +473,13 @@ $(function() {
...
+          $("#change-mobile").html("Whoops! Sorry , we don't support mobile devices.");
```
Comment (shakeelmohamed) [inline js/everything.js]: I don't like this id name, something like `#mobile-message` is better

File: js/everything.js [hunk 5]
```diff
@@ -474,7 +473,13 @@ $(function() {
...
+      });
+    }
```
Comment (shakeelmohamed) [inline js/everything.js]: The last statement inside this `if` should be a `return;` we don't want to try doing anything else if we're on mobile.
We should also make sure the demo button is hidden at this point since it won't work

File: js/everything.js (hunk 3 above)
Comment (FleetAdmiral) [inline js/everything.js]: Will be done sorry for the trouble 👍

**Comment by shakeelmohamed** [comment]
Left some more comments, please fix the style errors found on [Travis CI](https://travis-ci.org/zen-audio-player/zen-audio-player.github.io/jobs/136750025) as well

**Comment by FleetAdmiral** [comment]
@shakeelmohamed All changes made 👍

File: js/everything.js [hunk 6]
```diff
@@ -12,9 +12,6 @@ function anonymizeFileUrl() {
 }
 
 function sendKeenEvent(_msg, _data) {
-    if (!client) {
```
Comment (shakeelmohamed) [inline js/everything.js]: revert this change

File: js/everything.js [hunk 7]
```diff
@@ -430,13 +427,17 @@ function pickDemo() {
 }
 
 $(function() {
-    // Keen.io
-    if (typeof Keen !== "undefined") { // eslint-disable-line no-undef
```
Comment (shakeelmohamed) [inline js/everything.js]: revert this change

File: js/everything.js [hunk 8]
```diff
@@ -430,13 +427,17 @@ function pickDemo() {
...
-        client = new Keen({ // eslint-disable-line no-undef
-            projectId: "5690c384c1e0ab0c8a6c59c4",
-            writeKey: "630fa16847ce5ffb01c9cc00327498e4e7716e0f324fb14fdf0e83ffc06f9eacff5fad1313c2701efe4a91c88c34b8d8153cbb121c454056bb63caf60a46336dd9c9e9855ecc5202ef3151d798eda40896d5111f44005c707cbfb32c7ae31070d129d6f520d5604fdbce5ad31e9c7232"
-        });
+    if (/Android|webOS|iPhone|iPad|iPod|BlackBerry|IEMobile|Opera Mini/i.test(navigator.usersAgent))
```
Comment (shakeelmohamed) [inline js/everything.js]: After you revert the above change, move this `if ` statement above

File: js/everything.js [hunk 9]
```diff
@@ -430,13 +427,17 @@ function pickDemo() {
...
+    {
```
Comment (shakeelmohamed) [inline js/everything.js]: move this brace to end of the previous line

File: index.html [hunk 10]
```diff
@@ -39,11 +39,12 @@
 
         <hgroup>
             <h1>Zen Audio Player</h1>
-            <h2 class="color-grey">Listen to YouTube videos, without the distracting visuals.</h2>
+            <h2 class="color-grey" id="mobile-message">Listen to YouTube videos, without the distracting visuals.</h2>
         </hgroup>
```
Comment (shakeelmohamed) [inline index.html]: There should be another empty `h2` here with `id="mobile-message"`

Comment (shakeelmohamed) [inline index.html]: Then add a CSS rule in `css/styles.css`:

```css
#mobile-message {
    display: none;
}
```

File: js/everything.js [hunk 11]
```diff
@@ -430,13 +427,17 @@ function pickDemo() {
...
+        $("#container").hide();
+        $("#mobile-message").text("Whoops! Sorry , we don't support mobile devices.");
```
Comment (shakeelmohamed) [inline js/everything.js]: Change the string to `Sorry, we don't support mobile devices.`

File: js/everything.js [hunk 12]
```diff
@@ -430,13 +427,17 @@ function pickDemo() {
...
+        return;
     }
+    // Keen.io
```
Comment (shakeelmohamed) [inline js/everything.js]: Remove  this

**Comment by shakeelmohamed** [comment]
Looks like a merge went bad, left some comments

File: js/everything.js (hunk 8 above)
Comment (FleetAdmiral) [inline js/everything.js]: @shakeelmohamed Sorry , the new if statement and its body goes above the previous if statement ("if (typeof Keen !== "undefined") ") right?

Comment (shakeelmohamed) [inline js/everything.js]: yes
